addons/event/static/src/views/event_slot_calendar/event_slot_calendar_model.js

```diff
@@ -27,4 +40,27 @@ export class EventSlotCalendarModel extends CalendarModel {
         return normalizedRecord;
     }
 
+    /**
+     * @override
+     * Preserve slot duration when updating the start time by automatically adjusting the end time.
+     */
+    setMultiCreateTimeRange(timeRange) {
+        if (timeRange.start) {
+            const previousEndValues = this.getItemFromStorage("multiCreateTimeEnd", { hour: 13, minute: 0 });
+            const newStart = new Time(timeRange.start).toDateTime();
+            const previousStart = new Time(this.getItemFromStorage("multiCreateTimeStart", { hour: 12, minute: 0 })).toDateTime();
+            const previousEnd = new Time(previousEndValues).toDateTime();
+            const previousDelta = previousEnd.diff(previousStart, ['hours', 'minutes', 'seconds']).values;
+            const newEnd = newStart.plus({hour: previousDelta.hours, minute: previousDelta.minutes, second: previousDelta.seconds});
+            timeRange.end = new Time({
+                ...previousEndValues,
+                hour: newEnd.hour,
+                minute: newEnd.minute,
+                second: newEnd.second,
+            });
```

Comment:
probably missing something, but why do we need to do this (with a separate `previousEndValues` and using them here).  Are there any case this could be useful?